docs/tutorials/go/build-an-ecommerce-app/index.md

```diff
@@ -13,14 +13,690 @@ last_update:
+	var signal interface{}
+	c.Receive(ctx, &signal)
+
+	// Move your existing signals for adding or removing items from the cart to here
+})
+
+  // If the user doesn't update the cart for `abandonedCartTimeout`, send an email
+  // reminding them about their cart. Only send the email once.
+if !sentAbandonedCartEmail && len(state.Items) > 0 {
+	selector.AddFuture(workflow.NewTimer(ctx, abandonedCartTimeout), func(f workflow.Future) {
+		sentAbandonedCartEmail = true
+		ao := workflow.ActivityOptions{
+			StartToCloseTimeout:   10 * time.Second,
+		}
+
+		ctx = workflow.WithActivityOptions(ctx, ao)
+
+      // More on SendAbandonedCartEmail in the next section
+		err := workflow.ExecuteActivity(ctx, SendAbandonedCartEmail, state.Email).Get(ctx, nil)
+		if err != nil {
+			logger.Error("Error sending email %v", err)
+			return
+		}
+	})
+}
+
+selector.Select(ctx)
+}
+```
+
+You do not need to implement a job queue, write a separate Worker, or handle rescheduling jobs.
+All you need to do is create a new Selector after every Signal and use `selector.AddFuture()` to defer code that needs to happen after the associated timeout is selected.
+
+Temporal does the hard work of persisting and distributing the state of your Workflow for you.
+
+Next, let's take a closer look at Activities and the `ExecuteActivity()` call above that's responsible for sending the abandoned cart email.
+
+## Send emails from an Activity
+
+You can think of Activities as an abstraction for side effects in Temporal.
+[Workflows need to be idempotent functions](https://docs.temporal.io/workflows) to enable Temporal to re-run a Workflow to recreate the Workflow's state.
+Any side effects, like HTTP requests to the Mailgun API, should be in an Activity.
+
+Create another file named `activities.go` and define a function called `SendAbandonedCartEmail`.
+The function takes two parameters: the Workflow context and the email as a string.
+You'll also add your `MailgunDomain` and `MailgunKey` API params here, since your Activity is what contacts the Mailgun API. 
+
+```go
+package app
+
+import (
+	"context"
+	"fmt"
+	"github.com/mailgun/mailgun-go"
+)
+
+type Activities struct {
+	MailgunDomain string
+	MailgunKey    string
+}
+
+func (a *Activities) SendAbandonedCartEmail(_ context.Context, email string) error {
+	if email == "" {
+		return nil
+	}
+	mg := mailgun.NewMailgun(a.MailgunDomain, a.MailgunKey)
+	m := mg.NewMessage(
+		"noreply@"+a.MailgunDomain,
+		"You've abandoned your shopping cart!",
+		"Go to http://localhost:8080 to finish checking out!",
+		email,
+	)
+	_, _, err := mg.Send(m)
+	if err != nil {
+		fmt.Println("Mailgun err: " + err.Error())
+		return err
+	}
+
+	return err
+}
+```
+
+This way, when your Workflow calls `workflow.ExecuteActivity(ctx, SendAbandonedCartEmail, state.Email)`, you'll get all the benefits of Temporal [Activities](https://docs.temporal.io/activities), like a configurable [Retry Policy](https://docs.temporal.io/encyclopedia/retry-policies).
```

Comment:
Would benefit from what a retry policy is I think so there's not so many link click-aways

Reply:
That was originally here and I took it out because the tutorial was so long to begin with.